internal/gitlab/gitlab.go

```diff
@@ -874,6 +874,49 @@ func LabelDelete(project, name string) error {
 	return err
 }
 
+// BranchGet gets a specific named branch object from the project
+func BranchGet(project string, name string) (*gitlab.Branch, error) {
+	opts := &gitlab.ListBranchesOptions{
+		Search: &name,
+	}
+
+	branches, _ := BranchList(project, opts)
+	switch len(branches) {
+	case 1:
+		return branches[0], nil
+	case 0:
+		return nil, errors.Errorf("Branch '%s' not found", name)
+	default:
+		return nil, errors.Errorf("Branch '%s' is ambiguous", name)
```

Comment:
I think this needs something like the exact-match handling we do for label names. Otherwise it's impossible to set the target branch to `mainline` if a `mainline-next` branch exists as well, no?

Reply:
Hmmm.. indeed! Good catch. Will add the exact-match handling.